apis/v1alpha2/gateway_types.go

```diff
@@ -480,12 +552,12 @@ const (
 	// true.
 	ListenerReasonReady ListenerConditionReason = "Ready"
 
-	// This reason is used with the "Ready" condition when the
+	// This reason is used with the "Ready" and "Programmed" conditions when the
 	// Listener is syntactically or semantically invalid.
 	ListenerReasonInvalid ListenerConditionReason = "Invalid"
 
-	// This reason is used with the "Accepted" and "Ready" conditions when the
-	// Listener is either not yet reconciled or not yet not online and ready to
-	// accept client traffic.
+	// This reason is used with the "Accepted", "Ready" and "Programmed"
+	// conditions when the Listener is either not yet reconciled or not yet not
+	// online and ready to accept client traffic.
 	ListenerReasonPending ListenerConditionReason = "Pending"
```

Comment:
Same comment as above re: "Pending" vs. "Unknown"

Reply:
Again, I think that the _status_ should be `Unknown`, but the _reason_ can be `Pending`.